### ISSUE TITLE
Environment + vault

### ISSUE THREAD

**Issue by ric79** [ansible-semaphore/semaphore#285]
Hello, 
can I have more details about how to solve this problem? At the moment:

1)
**Extra CLI Arguments**
`["--vault-password-file", "vault-pass.py"]`

2)
vault-pass.py:
```
#!/usr/bin/python
import os
print os.environ['my_ansible_vault_pass']
```

3)
```
Environment JSON:
{
  "my_ansible_vault_pass": "xxxx"
}
```

However, I got:
```
Traceback (most recent call last):
  File "/home/ansible/playbooks/repository_1/vault-pass.py", line 3, in <module>
    print os.environ['my_ansible_vault_pass']
  File "/usr/lib/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'my_ansible_vault_pass'
ERROR! Vault password script /home/ansible/playbooks/repository_1/vault-pass.py returned non-zero (1): None
```
This means that semaphore is not setting the environment variable `my_ansible_vault_pass`. Any ideas?

Edit: Semaphore 2.2.0

**Comment by frostymarvelous** [comment]
From what I can tell, Environment doesn't set environment variables.
These are extra variables that are passed to Ansible. 

I could be wrong.

**Comment by matejkramny** [comment]
It is a file passed to ansible as `--extra-vars` parameter

**Comment by ric79** [comment]
I can be wrong, but extra-vars are not environment vars.

**Comment by matejkramny** [comment]
they're not really environment vars, you're right

**Comment by ykuksenko** [comment]
Also environment variables seem to be the only way to pass a custom ANSIBLE_CONFIG to ansible which is required if you want to use separate paths for fact caching between inventories that have the same named hosts. The primary key in ansible is the first column for the cache.
For example:
```
#inventory/production.yml
host1  ansible_host host1.production.domain.local
```
```
#inventory/development.yml
host1  ansible_host host1.development.domain.local
```
When you switch between these inventories ansible will have the wrong facts cached (like IP addresses). So currently you have to not cache at all or flush the cache when switching.

**Comment by fiftin** [comment]
Vault password can be configured in Template settings. Available in v2.7.20.

https://github.com/ansible-semaphore/semaphore/releases/tag/v2.7.20
